iris/tests/download_manager/change_download_folder.py

```diff
@@ -0,0 +1,160 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+
+from iris.test_case import *
+
+
+class Test(BaseTest):
+
+    def __init__(self):
+        BaseTest.__init__(self)
+        self.meta = 'Changing the download folder works accordingly.'
+        self.test_case_id = '99488'
+        self.test_suite_id = '1827'
+        self.locales = ['en-US']
+        self.blocked_by = {'id': '1513494', 'platform': [Platform.LINUX]}
+
+    def setup(self):
+        """Test case setup
+
+        This overrides the setup method in the BaseTest class, so that it can use a brand new profile.
+        """
+        BaseTest.setup(self)
+        self.profile = Profile.BRAND_NEW
+        self.set_profile_pref({'browser.download.dir': IrisCore.get_downloads_dir()})
+        self.set_profile_pref({'browser.download.folderList': 2})
+        self.set_profile_pref({'browser.download.useDownloadDir': True})
+        return
+
+    def run(self):
+        navigate(LocalWeb.THINKBROADBAND_TEST_SITE)
+
+        # Wait for the page to be loaded.
+        try:
+            wait(DownloadFiles.EXTRA_SMALL_FILE_5MB, 10)
+            logger.debug('File is present in the page.')
+        except FindError:
+            raise FindError('File is not present in the page.')
+
+        download_file(DownloadFiles.EXTRA_SMALL_FILE_5MB, DownloadFiles.OK)
+
+        expected = exists(NavBar.DOWNLOADS_BUTTON_BLUE, 10)
+        assert_true(self, expected, 'Downloads button found.')
+
+        click(NavBar.HOME_BUTTON.target_offset(70, 0))
+
+        # Navigate to about:preferences.
+        navigate('about:preferences#searchResults')
+
+        expected = exists(AboutPreferences.ABOUT_PREFERENCE_SEARCH_PAGE_PATTERN, 10)
+        assert_true(self, expected, 'The \'about:preferences#search\' page successfully loaded.')
+
+        time.sleep(DEFAULT_UI_DELAY_LONG)
```

Comment:
> Please remove the time.sleep() from line 54. I don't think that we need it.

When is running too fast, we will miss some characters from the next step: `type('downloads')`, like in the image attached, having less characters we may not reach to the Choose/Browse button from Downloads section. Is mandatory to use this here in order to avoid this type of fail
![2019_04_24_18_09_55_320264_change_download_folder](https://user-images.githubusercontent.com/35529471/56680248-9a082800-66cf-11e9-8f6a-6421f7396e83.jpg)
.

Reply:
This is because you use type('downloads') instead of paste('downloads')

Reply:
I was using type because is already in use in other tests, there is nowhere specify to use paste() instead type() as a better practice

Reply:
even using paste(), won't gonna work without a small time sleep, to avoid using a time sleep here would be required to look for a pattern on the Search section other than the assert on the URL

Reply:
We should use in this case paste() and assert for a pattern from the opened page.